sui-execution/latest/sui-adapter/src/temporary_store.rs

```diff
@@ -196,14 +197,30 @@ impl<'backing> TemporaryStore<'backing> {
         mut self,
         shared_object_refs: Vec<ObjectRef>,
         transaction_digest: &TransactionDigest,
-        transaction_dependencies: Vec<TransactionDigest>,
+        mut transaction_dependencies: BTreeSet<TransactionDigest>,
         gas_cost_summary: GasCostSummary,
         status: ExecutionStatus,
         gas_charger: &mut GasCharger,
         epoch: EpochId,
     ) -> (InnerTemporaryStore, TransactionEffects) {
         let mut modified_at_versions = vec![];
 
+        // Regardless of execution status (including aborts), we insert the previous transaction
+        // for any successfully received objects during the transaction.
+        for (id, expected_version, expected_digest) in self.receiving_objects.iter() {
+            // If the receiving object is in the loaded runtime objects, then that means that it
+            // was actually successfully loaded (so existed, and there was authenticated mutable
+            // access to it). So we insert the previous transaction as a dependency.
+            if let Some(obj_meta) = self.loaded_runtime_objects.get(id) {
+                // Check that the expected version and digest match the loaded version
+                // and digest. If they don't then don't register a dependency.
+                // This is because this could be "spoofed" by loading a dynamic object field.
+                if obj_meta.version == *expected_version && obj_meta.digest == *expected_digest {
```

Comment:
Would someone still be able to spoof it here, by actually loading a child object at id and version specified as a receiving object? Would that lead to the registration of the dependency?

Reply:
Ah good question -- this is not possible since in order to load an object via dynamic object fields it will have an `ObjectOwner` ownership, but in order to load it via `receiving` that object at that same version must have an `AddressOwner` relationship.

Since we check that there is an `AddressOwner` ownership of the object (or that it was already received at that version) at signing this would then mean that the object at the same version must have had both `ObjectOwner` ownership and `AddressOwner` ownership which isn't possible. So this cannot happen. 

Additionally, even if this did somehow become the case and we did somehow register a dependency here for a dynamic object field load, it wouldn't effect anything since it would be registering a transaction dependency that must be less than or equal to any transaction dependencies added from the parent object of the dynamic field that must also be present in the transaction.